clang/lib/Sema/SemaDecl.cpp

```diff
@@ -3215,8 +3215,8 @@ static void adjustDeclContextForDeclaratorDecl(DeclaratorDecl *NewD,
 template <typename AttributeType>
 static void checkDimensionsAndSetDiagnostics(Sema &S, FunctionDecl *New,
                                              FunctionDecl *Old) {
-  AttributeType *NewDeclAttr = New->getAttr<AttributeType>();
-  AttributeType *OldDeclAttr = Old->getAttr<AttributeType>();
+  const auto *NewDeclAttr = New->getAttr<AttributeType>();
```

Comment:
I think you need to add an assert or an if statement. @erichkeane  what is the 'correct' way to fix these issues?

Reply:
Based on its use, it appears that both uses do a 'hasAttr' on both of those.  I have a slight preference however for THIS to do the 'if' statement (and early return if !NewDeclAttr || !OldDeclAttr), and change https://github.com/intel/llvm/blob/00693a615eb35137d1fa5993f7b037de82f61dd2/clang/lib/Sema/SemaDecl.cpp#L3305 and 3309 to just call it without their 'if' statements.

In this case, we could also get away with just an 'assert', but that seems like worse code.

Reply:
hanks Erich and Elizabeth for the review and suggestion. I have updated patch.